lib/netconf/netconf.go

```diff
@@ -77,14 +77,15 @@ func (n Network) Chain(id uint64) (Chain, bool) {
 // the Omni cross chain protocol. This is most supported Rollup EVM, but
 // also the Omni EVM.
 type Chain struct {
-	ID            uint64        // Chain ID asa per https://chainlist.org
-	Name          string        // Chain name as per https://chainlist.org
-	RPCURL        string        // RPC URL of the chain
-	AuthRPCURL    string        // RPC URL of the chain with JWT authentication enabled
-	PortalAddress string        // Address of the omni portal contract on the chain
-	DeployHeight  uint64        // Height that the portal contracts were deployed
-	IsOmni        bool          // Whether this is the Omni chain
-	BlockPeriod   time.Duration // Block period of the chain
+	ID              uint64        // Chain ID asa per https://chainlist.org
+	Name            string        // Chain name as per https://chainlist.org
+	RPCURL          string        // RPC URL of the chain
+	AuthRPCURL      string        // RPC URL of the chain with JWT authentication enabled
+	PortalAddress   string        // Address of the omni portal contract on the chain
+	DeployHeight    uint64        // Height that the portal contracts were deployed
+	IsOmni          bool          // Whether this is the Omni chain
+	BlockPeriod     time.Duration // Block period of the chain
+	CommitmentLevel string        // Commitment Level of the block ("latest", "finalized")
```

Comment:
"latest" "finalised" might not be enough for rollups like ZKrollups where we will need a completely different strategy.

So rather name this `FinalizationStrat`.

Also please make this an `string enum` in netconf with two values `StratLatest` and `StratFinalized`